services/storage/src/simcore_service_storage/db_file_meta_data.py

```diff
@@ -72,7 +72,7 @@ def _list_filter_with_partial_file_id_stmt(
     file_id_prefix: str | None,
     partial_file_id: str | None,
     sha256_checksum: SHA256Str | None,
-    only_files: bool,
+    is_directory: bool | None = None,
```

Comment:
Q: why put a default here? this is a private function, there is no need.

Reply:
None is to not apply the condition: both files and dirs

Reply:
Yes I get that what I meant is the default value that you apply here. Why do you set a default? this does not need to be backward compatible.

Reply:
Because when I invoke this function I don't want to specify so much stuff. Now we have a 3rd option: `None` == _no filter condition_ in addition to `True/False` used to filter by this column...

Reply:
I still don't get it.
You replaced `only_files` which was a boolean that was a required argument that is only used once in this module with `is_directory` which is now a nullable boolean that is now not required because it is defaulted to `None`. So you anyway need to change everywhere you use that function because `only_files` disappeared (e.g. the syntax changed so or so).